pypac/learner.py

```diff
@@ -59,3 +60,54 @@ def low_degree_chi(self, degree):
         """
         for indices in combinations(range(self.instance.input_length), degree):
             yield [1 if i in indices else 0 for i in range(self.instance.input_length)]
+
+
+class GoldreichLevin:
+    """
+    Probabilistic algorithm to return a list of sets for the `instance` Boolean function using query access.
+    If the magnitude of a coefficient is greater or equal than `tau` its set is guaranteed to be in the output list.
+    On the other hand all sets in the output list are guaranteed to have Fourier coefficient magnitude greater or equal
+    than 1/2 `tau`.
+    """
+
+    def __init__(self, instance, tau, delta):
+        self.instance = instance
+        self.tau = tau
+        epsilon = tau ** 2 / 4
+        self.delta = tau ** 2 / (8 * self.instance.input_length) / delta
+        self.sample_size = math.ceil(-math.log(self.delta) * (2 + epsilon) / (epsilon ** 2))
+
+    def learn(self):
```

Comment:
This function should not be called `learn` as it is not a PAC learning algorithm. Actually, maybe GoldreichLevin should be moved to `tools.py` or be encapsulated into a `KushilevitzMansourAlgorithm` class, that provides a `learn` method. Inferring from `LowDegree`, the user would expect a hypothesis as return of `learn()`.